syntax = "proto3";
package service;

service Homomorphic {
    rpc Compute(ComputationRequest) returns (ComputationResponse) {}
}

message EncryptedData {
    repeated int64 polynomials0 = 1;
    repeated int64 polynomials1 = 2;
}

message ComputationRequest {
    repeated EncryptedData data_array = 1;
    string operation = 2;
    string evaluation_key = 3;
}

message ComputationResponse {
    int64 code = 1;
    string message = 2;
}